code-examples/core_features/web3names/1_web3name.ts

```diff
@@ -0,0 +1,66 @@
+import { ApiPromise } from '@polkadot/api'
+import { randomAsHex } from '@polkadot/util-crypto'
+import { KeyringPair } from '@kiltprotocol/types'
+
+import {
+  DemoKeystore,
+  FullDidDetails,
+  SigningAlgorithms,
+  FullDidCreationBuilder,
+  Web3Names
+} from '@kiltprotocol/did'
+import {
+  SubscriptionPromise,
+  VerificationKeyType,
+  NewDidVerificationKey
+} from '@kiltprotocol/types'
+import { BlockchainUtils } from '@kiltprotocol/chain-helpers'
+
+export async function main(
+  api: ApiPromise,
+  keystore: DemoKeystore,
+  kiltAccount: KeyringPair,
+  web3Name: Web3Names.Web3Name,
+  resolveOn: SubscriptionPromise.ResultEvaluator = BlockchainUtils.IS_FINALIZED
+): Promise<FullDidDetails> {
+  // Generate a random new full DID.
+  const authenticationSeed = randomAsHex(32)
+
+  // Ask the keystore to generate a new keypair to use for authentication with the generated seed.
+  const authenticationKeyPublicDetails: NewDidVerificationKey = await keystore
+    .generateKeypair({
+      alg: SigningAlgorithms.Ed25519,
+      seed: authenticationSeed
+    })
+    .then((key) => {
+      return {
+        publicKey: key.publicKey,
+        type: VerificationKeyType.Ed25519
+      }
+    })
+
+  const fullDid = await new FullDidCreationBuilder(
+    api,
+    authenticationKeyPublicDetails
+  ).consumeWithHandler(keystore, kiltAccount.address, async (tx) => {
+    await BlockchainUtils.signAndSubmitTx(tx, kiltAccount, {
+      reSign: true,
+      resolveOn
+    })
+  })
+  console.log(fullDid.did)
+
+  // Claim the Web3 name if it has not been claimed by anyone else.
+  const doesNameExist = await Web3Names.queryDidForWeb3Name(web3Name)
+  if (!doesNameExist) {
+    const web3NameClaimTx = await Web3Names.getClaimTx(web3Name).then((tx) =>
+      fullDid.authorizeExtrinsic(tx, keystore, kiltAccount.address)
+    )
+    await BlockchainUtils.signAndSubmitTx(web3NameClaimTx, kiltAccount, {
+      reSign: true,
+      resolveOn
+    })
+  }
```

Comment:
Not sure if we want to include these kind of check.
I think we should keep it as short as possible and therefore remove these checks. The code might not be production ready than, but it's easier to understand.

```suggestion
  const web3NameClaimTx = await Web3Names.getClaimTx(web3Name).then((tx) =>
    fullDid.authorizeExtrinsic(tx, keystore, kiltAccount.address)
  )
  await BlockchainUtils.signAndSubmitTx(web3NameClaimTx, kiltAccount, {
    reSign: true,
    resolveOn
  })
```

Reply:
Yeah, I was also reluctant to add these checks, but these code snippets are tested on new PRs, and they would fail if some bit of data is created deterministically. So for web3 names, we either create a random web3 name every time, or we need to check that what is being created does not exist yet. Perhaps the checks about the owner can be removed, agree, but what about existential checks?

Reply:
maybe we could move the check to the `index.ts` file? isn't the w3n deleted after each test? So even if it's deterministic, it's free again after the test?

Reply:
I'll try to put it into the `index.ts` file, that's a good suggestion. Anyway, we can't make assumptions since we are using Peregrine and it is definitely possible that someone else would claim the same web3 name that is used in the example. But checking in the index file should be enough.

Reply:
https://github.com/KILTprotocol/docs/pull/62/commits/dc9c6f81948c41c29dbc56979e45e2c8bcf9d300